frontend/gateway/gateway.go

```diff
@@ -104,7 +104,12 @@ func (gf *gatewayFrontend) Solve(ctx context.Context, llbBridge frontend.Fronten
 		if !ok {
 			return nil, errors.Errorf("invalid ref: %T", res.Sys())
 		}
-		rootFS = workerRef.ImmutableRef
+
+		rootFS, err = workerRef.Worker.CacheManager().New(ctx, workerRef.ImmutableRef)
+		if err != nil {
+			return nil, err
+		}
+		defer rootFS.Release(ctx)
```

Comment:
the `Release` methods need `context.TODO()` as when context gets canceled we still need to release the refs.

Reply:
I see, thanks for the comment. Fixed.